src/pages/editor/curators/AddCuratorDialog.tsx

```diff
@@ -0,0 +1,183 @@
+import { AutocompleteTextField } from '../../../components/AutocompleteTextField';
+import { RootState } from '../../../redux/store';
+import { Organization } from '../../../types/organization.types';
+import { CristinPerson, InstitutionUser } from '../../../types/user.types';
+import { dataTestId } from '../../../utils/dataTestIds';
+import { useDebounce } from '../../../utils/hooks/useDebounce';
+import { getAllChildOrganizations } from '../../../utils/institutions-helpers';
+import { getFullCristinName, getUsername } from '../../../utils/user-helpers';
+import { AddCuratorForm } from './AddCuratorForm';
+import { OrganizationCuratorsAccordionProps } from './OrganizationCuratorsAccordion';
+
+interface AddCuratorDialogProps
+  extends Pick<DialogProps, 'open'>,
+    Pick<OrganizationCuratorsAccordionProps, 'refetchCurators' | 'parentOrganizationIds'> {
+  onClose: () => void;
+  currentOrganization: Organization;
+}
+
+export const AddCuratorDialog = ({
+  onClose,
+  open,
+  currentOrganization,
+  refetchCurators,
+  parentOrganizationIds,
+}: AddCuratorDialogProps) => {
+  const { t } = useTranslation();
+  const user = useSelector((store: RootState) => store.user);
+  const topOrgCristinId = user?.topOrgCristinId ?? '';
+
+  const [searchQuery, setSearchQuery] = useState('');
+  const debouncedSearchQuery = useDebounce(searchQuery);
+
+  const [selectedPerson, setSelectedPerson] = useState<CristinPerson | null>(null);
+  const [userInitialValues, setUserInitialValues] = useState<InstitutionUser | null>(null);
+
+  const closeDialog = () => {
+    setSearchQuery('');
+    setSelectedPerson(null);
+    setUserInitialValues(null);
+    onClose();
+  };
+
+  const employeeSearchQuery = useQuery({
+    enabled: open && !!topOrgCristinId && !!debouncedSearchQuery && debouncedSearchQuery === searchQuery,
+    queryKey: ['employees', topOrgCristinId, 20, 1, debouncedSearchQuery],
+    queryFn: ({ signal }) => fetchEmployees(topOrgCristinId, 20, 1, debouncedSearchQuery, signal),
+    meta: { errorMessage: t('feedback.error.get_users_for_institution') },
+  });
+
+  const username = getUsername(selectedPerson, topOrgCristinId);
+
+  const userQuery = useQuery({
+    enabled: open && !!selectedPerson && !!username,
+    queryKey: ['user', username],
+    queryFn: () => fetchUser(username),
+    meta: { errorMessage: false }, // No error message, since a Cristin Person will lack User if they have not logged in yet
+    retry: false,
+    cacheTime: 0, // Disable caching since this user in many cases will be changed, and the cached data then will be outdated
+  });
+
+  useEffect(() => {
+    const currentUser = userQuery.data;
+    if (currentUser) {
+      let viewingScope = currentUser.viewingScope.includedUnits;
+
+      const newOrganizationId = !viewingScope.includes(currentOrganization.id) ? currentOrganization.id : null;
+      if (newOrganizationId) {
+        // Remove organizations conflicting with the new organization
```

Comment:
Denne logikken med å fjerne overlapp når man legger til ansvarsområder, burde det trekkes ut i en hjelpefunksjon? Om det er andre plasser som endrer på ansvarsområder så hadde det vært kjekt om man viderebruker. Eller er dette kanskje eneste plass dette skjer?

Reply:
Jepp, har tenkt tanken, men tenker å vente til det samme skal brukes i endre person også.
Ser for meg det blir en funksjon som kan ta inn nåværende liste, og ny liste (eller element?), og som returnerer ny liste, muligens også med markering om hva som er fjernet og lagt til 🤔